### PR TITLE
Add adapt_structure to the CAN parameters to allow GPU usage

### PR DESCRIPTION
**Pull request by hengdiliang**
## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/carbon_alkalinity_nutrients.jl [hunk 1]
```diff
@@ -222,16 +222,16 @@ end
...
+Calculate the remineralization of dissolved organic phosphorus.
 """
-@inline dissolved_organic_phosphate_remin(remineralization_rate, 
+@inline dissolved_organic_phosphorus_remin(remineralization_rate, 
                                           dissolved_organic_phosphorus_concentration) = 
         max(0, remineralization_rate * dissolved_organic_phosphorus_concentration)
 
 # Martin Curve
-@inline particulate_organic_phosphate_remin() = 0.0
+@inline particulate_organic_phosphorus_remin() = 0.0
```
Comment (glwagner) [inline src/carbon_alkalinity_nutrients.jl]:
```suggestion
@inline particulate_organic_phosphorus_remin() = 0
```

doesn't matter at all, but its good just to build a habit of not hardcoding Float64

Comment (hengdiliang) [reply src/carbon_alkalinity_nutrients.jl]: Got it

**Review by seamanticscience — APPROVED** [review]
Looks good to me! LGTM!

**Comment by glwagner** [comment]
Was this tested? How do you test on GPU?

**Pull request merged**